medmodels/medrecord/overview.py

```diff
@@ -0,0 +1,135 @@
+from medmodels.medrecord.types import Attributes, EdgeIndex, NodeIndex
+
+
+def extract_attribute_summary(
+    attribute_dict: Union[Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]],
+    schema: Optional[AttributesSchema] = None,
+    decimal: Optional[int] = 2,
+) -> pl.DataFrame:
+    """Extracts a summary from a node or edge attribute dictionary.
+
+    Args:
+        attribute_dict (Union[Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]]):
+             Edges or Nodes and their attributes and values.
+        schema (Optional[AttributesSchema], optional): Attribute Schema for the group
+            nodes or edges. Defaults to None.
+        decimal (Optional[int], optional): Decimal points to round the numeric values.
+            Defaults to 2.
+
+    Returns:
+        pl.DataFrame: Summary of node or edge attributes.
+    """
+
+    data = pl.DataFrame(data=[{"id": k, **v} for k, v in attribute_dict.items()])
+
+    data_dict = {
+        "Attribute": [],
+        "Info": [],
+    }
+
+    attributes = [col for col in data.columns if col != "id"]
+    attributes.sort()
+    if attributes:
+        for attribute in attributes:
+            if schema and attribute in schema:
+                if schema[attribute][1] == AttributeType.Continuous:
+                    attribute_info = [
+                        f"min: {data[attribute].min()}",
+                        f"max: {data[attribute].max()}",
+                        f"mean: {data[attribute].mean():.{decimal}f}",
+                    ]
+
+                elif schema[attribute][1] == AttributeType.Temporal:
+                    attribute_info = [
+                        f"min: {min(data[attribute])}",
+                        f"max: {max(data[attribute])}",
+                    ]
+
+                elif schema[attribute][1] == AttributeType.Categorical:
+                    categories = data[attribute].unique().sort()
+                    if len(categories) > 5:
+                        attribute_info = [f"{len(categories)} categories"]
+                    else:
+                        category_str = ", ".join(categories)
+                        if len(category_str) < 100:
+                            attribute_info = [
+                                f"Categories: {', '.join(list(categories))}"
+                            ]
+                        else:
+                            attribute_info = [f"{len(categories)} categories"]
+                else:
+                    attribute_info = [f"{len(data[attribute].unique())} unique values"]
+
+            ## Without Schema
+            else:
+                if data[attribute].dtype.is_numeric():
+                    attribute_info = [
+                        f"min: {data[attribute].min()}",
+                        f"max: {data[attribute].max()}",
+                        f"mean: {data[attribute].mean():.{decimal}f}",
+                    ]
+                else:
+                    try:
```

Comment:
`try ... except` is bad practice when other options are available. Try using `data[attribute].str.to_datetime(strict=False)` for example which would not raise an error

Reply:
data[attribute].str.to_datetime(strict=False) does not work in the case of not having any datetime object in the series.